apps/smart-contracts/core/test/FeeReimbursement.test.ts

```diff
@@ -0,0 +1,186 @@
+    })
+
+    it('sets DEFAULT_ADMIN_ROLE holder to deployer', async () => {
+      expect(await feeReimbursement.hasRole(DEFAULT_ADMIN_ROLE, deployer.address)).to.eq(true)
+    })
+
+    it('sets role constants to the correct hash', async () => {
+      expect(await feeReimbursement.SET_DEPOSIT_HOOK_ROLE()).to.eq(
+        id('FeeReimbursement_setDepositHook(IDepositHook)')
+      )
+      expect(await feeReimbursement.SET_MINISALES_ROLE()).to.eq(
+        id('FeeReimbursement_setMiniSales(IMiniSales)')
+      )
+      expect(await feeReimbursement.SET_PPO_TOKEN_ROLE()).to.eq(
+        id('FeeReimbursement_setPPOToken(IERC20)')
+      )
+    })
+  })
+
+  describe('# registerFee', () => {
+    beforeEach(async () => {
+      await setupFeeReimbursement()
+    })
+
+    it('should be callable by connected deposit hook', async () => {
+      await feeReimbursement.connect(depositHook).registerFee(user.address, TEST_AMOUNT_ONE)
+    })
+
+    it('should not be callable by non-authorised address', async () => {
+      const tx = feeReimbursement.connect(user).registerFee(user.address, TEST_AMOUNT_ONE)
+      await expect(tx).to.be.revertedWith('msg.sender != depositHook')
+    })
+
+    it("should correctly add 'amount' to fee", async () => {
+      const feesPaidBefore = await feeReimbursement.getFeesPaid(user.address)
+      expect(feesPaidBefore).to.eq(0)
+      await feeReimbursement.connect(depositHook).registerFee(user.address, TEST_AMOUNT_ONE)
+      const feesPaidAfter = await feeReimbursement.getFeesPaid(user.address)
+      expect(feesPaidAfter).to.eq(TEST_AMOUNT_ONE)
+    })
+  })
+
+  describe('# getClaimablePPO', () => {
+    beforeEach(async () => {
+      await setupFeeReimbursement()
+    })
+
+    it('should return the correct amount of claimable PPO for fees paid', async () => {
+      await feeReimbursement.connect(depositHook).registerFee(user.address, TEST_AMOUNT_ONE)
+      const feesPaid = await feeReimbursement.getFeesPaid(user.address)
+      expect(feesPaid).to.eq(TEST_AMOUNT_ONE)
+      const claimablePPO = await feeReimbursement.getClaimablePPO(user.address)
+      expect(claimablePPO).to.eq(TEST_AMOUNT_ONE)
+    })
+  })
+
+  describe('# claim', () => {
+    beforeEach(async () => {
+      await setupFeeReimbursement()
+    })
+
+    it('reverts if no fee pending', async () => {
+      const claimablePPO = await feeReimbursement.getClaimablePPO(user.address)
+      expect(claimablePPO).to.eq(0)
+      const tx = feeReimbursement.connect(user).claim()
+      await expect(tx).to.be.revertedWith('No reimbursement available')
+    })
+
+    it('claims correct amount of PPO if pending fee > 0', async () => {
+      await feeReimbursement.connect(depositHook).registerFee(user.address, TEST_AMOUNT_ONE)
+      await feeReimbursement.connect(user).claim()
+      const ppoBalance = await ppoToken.balanceOf(user.address)
+      expect(ppoBalance).to.eq(TEST_AMOUNT_ONE)
+    })
+
+    it('reverts if claiming twice', async () => {
```

Comment:
```suggestion
    it('reverts if claiming again and no fees accrued', async () => {
```
This should be clearer because this doesnt necessarily have anything to do with just calling the function twice.